dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java

```diff
@@ -315,9 +323,9 @@ public void saveInputStream(InputStream inputStream) throws IOException {
         
         File tempFile = copyInputStreamToTempFile(inputStream);
         long tempFileSize = tempFile.length();
-
+        String tempFileChecksum = FileUtil.calculateChecksum(tempFile.getAbsolutePath(), ChecksumType.MD5);
```

Comment:
In this case we're not checking this with datafile database checksum value. I propose to get rid of:
```
    public abstract void saveInputStream(InputStream inputStream) throws IOException;

    public abstract void saveInputStream(InputStream inputStream, Long filesize) throws IOException;
```
They most likely are not used.